Makefile

```diff
@@ -113,3 +107,71 @@ generate-pip-lockfiles: auth/pinned-requirements.txt
 generate-pip-lockfiles: batch/pinned-requirements.txt
 generate-pip-lockfiles: ci/pinned-requirements.txt
 generate-pip-lockfiles: memory/pinned-requirements.txt
+
+$(HAILTOP_VERSION):
+	$(MAKE) -C hail python/hailtop/hail_version
+
+hail-ubuntu-image: docker/hail-ubuntu
+	$(eval HAIL_UBUNTU_IMAGE := $(DOCKER_PREFIX)/hail-ubuntu:$(TOKEN))
+	python3 ci/jinja2_render.py '{"global":{"docker_prefix":"$(DOCKER_PREFIX)"}}' docker/hail-ubuntu/Dockerfile docker/hail-ubuntu/Dockerfile.out
+	./docker-build.sh docker/hail-ubuntu Dockerfile.out $(HAIL_UBUNTU_IMAGE)
+	echo $(HAIL_UBUNTU_IMAGE) > $@
+
+base-image: hail-ubuntu-image docker/Dockerfile.base
```

Comment:
Ya we'd still need some mapping in the Makefile for FROM images. I think it would be cleaner to use docker build-args and require all of our Dockerfiles start with `ARG BASE_IMAGE \ FROM ${BASE_IMAGE}` instead of jinja templating, but that doesn't solve the parametrization issue.